File: libs/shared/models/typescript/workout.ts
/**
 * Shared TypeScript interfaces for workout routines and exercises.
 */

export interface ExerciseBase {
  name: string;
  category: 'strength' | 'cardio' | 'flexibility';
  muscle_groups: string[];
  equipment?: string;
}

export interface ExerciseCreate extends ExerciseBase {}

export interface ExerciseResponse extends ExerciseBase {
  id: number;
  is_custom: boolean;
  created_at: string;
}

export interface RoutineExerciseBase {
  exercise_id: number;
  target_sets: number;
  target_reps: number;
  order: number;
}

export interface RoutineExerciseCreate extends RoutineExerciseBase {}

export interface RoutineExerciseResponse extends RoutineExerciseBase {
  id: number;
  exercise: ExerciseResponse;
}

export interface RoutineBase {
  name: string;
  description?: string;
}

export interface RoutineCreate extends RoutineBase {
  exercises: RoutineExerciseCreate[];
}

export interface RoutineUpdate {
  name?: string;
  description?: string;
  exercises?: RoutineExerciseCreate[];
}

export interface RoutineResponse extends RoutineBase {
  id: number;
  created_at: string;
  updated_at: string;
}

export interface RoutineDetailResponse extends RoutineResponse {
  exercises: RoutineExerciseResponse[];
}

File: libs/shared/models/kotlin/Workout.kt
package com.sentinel.shared.models

import java.time.Instant

/**
 * Shared Kotlin data classes for workout routines and exercises.
 */

data class ExerciseBase(
    val name: String,
    val category: String, // strength|cardio|flexibility
    val muscle_groups: List<String>,
    val equipment: String? = null
)

data class ExerciseCreate(
    val name: String,
    val category: String,
    val muscle_groups: List<String>,
    val equipment: String? = null
)

data class ExerciseResponse(
    val id: Int,
    val name: String,
    val category: String,
    val muscle_groups: List<String>,
    val equipment: String?,
    val is_custom: Boolean,
    val created_at: Instant
)

data class RoutineExerciseBase(
    val exercise_id: Int,
    val target_sets: Int,
    val target_reps: Int,
    val order: Int
)

data class RoutineExerciseCreate(
    val exercise_id: Int,
    val target_sets: Int,
    val target_reps: Int,
    val order: Int
)

data class RoutineExerciseResponse(
    val id: Int,
    val exercise_id: Int,
    val exercise: ExerciseResponse,
    val target_sets: Int,
    val target_reps: Int,
    val order: Int
)

data class RoutineBase(
    val name: String,
    val description: String? = null
)

data class RoutineCreate(
    val name: String,
    val description: String? = null,
    val exercises: List<RoutineExerciseCreate> = emptyList()
)

data class RoutineUpdate(
    val name: String? = null,
    val description: String? = null,
    val exercises: List<RoutineExerciseCreate>? = null
)

data class RoutineResponse(
    val id: Int,
    val name: String,
    val description: String?,
    val created_at: Instant,
    val updated_at: Instant
)

data class RoutineDetailResponse(
    val id: Int,
    val name: String,
    val description: String?,
    val created_at: Instant,
    val updated_at: Instant,
    val exercises: List<RoutineExerciseResponse> = emptyList()
)
